processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
The point of requesting the benchmark was to see the effects of having both code paths operating inside the same runtime.  I.e. if JIT were to optimize assuming it's always v1, introducing a code path that uses v2 will cause it to undo some of its optimizations.  The request for the benchmark is to understand if there is any difference between the code if both versions are present in the same runtime.

Given that the goal is to understand the effects of multiple code paths on the same runtime, using that parameter to force it to have both of them present should be totally valid.